packages/maskbook/src/plugins/PoolTogether/utils.ts

```diff
@@ -0,0 +1,94 @@
+import { ChainId } from '@masknet/web3-shared'
+import { ONE_SECOND, ONE_WEEK_SECONDS } from './constants'
+import type { Pool } from './types'
+
+export const subtractDates = (dateA: Date, dateB: Date) => {
+    // https://github.com/pooltogether/pooltogether-community-ui/blob/master/lib/utils/subtractDates.js
+
+    let msA = dateA.getTime()
+    let msB = dateB.getTime()
+
+    let diff = msA - msB
+
+    let days = 0
+    if (diff >= 86400000) {
+        days = Math.floor(diff / 86400000)
+        diff -= days * 86400000
+    }
+
+    let hours = 0
+    if (days || diff >= 3600000) {
+        hours = Math.floor(diff / 3600000)
+        diff -= hours * 3600000
+    }
+
+    let minutes = 0
+    if (hours || diff >= 60000) {
+        minutes = Math.floor(diff / 60000)
+        diff -= minutes * 60000
+    }
+
+    let seconds = 0
+    if (minutes || diff >= 1000) {
+        seconds = Math.floor(diff / 1000)
+    }
+
+    return {
+        days,
+        hours,
+        minutes,
+        seconds,
+    }
+}
+
+export const calculateOdds = (usersTicketBalance: number, totalSupply: number, numberOfWinners: number) => {
+    if (usersTicketBalance === 0 || isNaN(usersTicketBalance)) {
+        return undefined
+    }
+
+    // Calculate odds of winning at least 1 of the possible scenarios.
+    // 1/N, 2/N ... N-1/N, N/N
+    // Then we always display "1 in ____" so 1 / X.
+    const odds = 1 / (1 - Math.pow(totalSupply / (totalSupply + usersTicketBalance), numberOfWinners))
+
+    return odds.toLocaleString(undefined, {
+        maximumFractionDigits: odds >= 10 ? 0 : 8,
+    })
+}
+
+export const calculateNextPrize = (pool: Pool) => {
+    const rawPrize =
+        Number.parseInt(pool.prize.weeklyTotalValueUsd, 10) /
+        (ONE_WEEK_SECONDS / Number.parseInt(pool.config.prizePeriodSeconds, 10))
+
+    let formatedPrize
+    if (!isNaN(rawPrize)) {
+        formatedPrize = `\$${rawPrize.toLocaleString(undefined, { maximumFractionDigits: 0 })}`
+    } else {
+        const pirzeAmount = Number.parseFloat(pool.prize.amount)
+        formatedPrize = `${pirzeAmount.toLocaleString(undefined, {
+            maximumFractionDigits: pirzeAmount >= 10 ? 0 : 2,
+        })} ${pool.tokens.underlyingToken.symbol}`
+    }
+    return formatedPrize
+}
+
+export const calculateSecondsRemaining = (pool: Pool) => {
+    return (
+        Number.parseInt(pool.prize.prizePeriodStartedAt.hex, 16) +
+        Number.parseInt(pool.prize.prizePeriodSeconds.hex, 16) -
+        new Date().getTime() / ONE_SECOND
+    )
```

Comment:
why not the parsed value as variable?

```suggestion
    return started_at + seconds - (Date.now() / ONE_SECOND)
```

see https://mdn.io/Date.now